presc/copies/continuous.py

```diff
@@ -0,0 +1,185 @@
+            data_block = self.classifier_copy.generate_synthetic_data(
+                random_state=self.random_state
+            )
+            self.random_state = self.random_state + 1
+            self.data_stream.put(data_block)
+
+    def stop(self):
+        self.data_streamer_run = False
+        if self.verbose:
+            print("Stopping data streamer...\n")
+
+
+class ContinuousCopy(Thread):
+    """Continuous classifier copy for online classifiers.
+
+    This class allows to instantiate online classifiers to perform sequential
+    incremental training of the classifier copy, which can be carried out
+    continuously non-stop. It keeps taking the necessary data batches from the
+    specified queue as soon as they are available, but it is agnostic as to how
+    the data is added to the queue.
+
+    Attributes
+    ----------
+        classifier_copy : presc.copies.copying.ClassifierCopy
+            An instance of the ClassifierCopy class with the original,
+            classifier, its associated sampling method of choice, and the online
+            copy classifier, which needs to be capable of incremental training.
+        data_stream : queue.Queue
+            The queue where the batches of data for the incremental training of
+            the online copy can be found.
+        fit_kwargs : dict or dict of dicts
+            A dictionary with the fit parameters needed for the online copy
+            classifier, or for each element of the online copy pipeline. In the
+            latter case, there must be an entry for each element of the pipeline
+            and each one in turn must have a dictionary with its fit parameters.
+        verbose : bool
+            If True it provides details at each iteration of the number of
+            samples used so far for the incremental training of the copy, and
+            evaluates the performance of the resulting copy.
+        test_data : presc.dataset.Dataset
+            Dataset with the test data that will be used to evaluate the copy
+            after each training iteration.
+    """
+
+    def __init__(
+        self, classifier_copy, data_stream, fit_kwargs={}, verbose=False, test_data=None
+    ):
+        Thread.__init__(self)
+        self.classifier_copy = classifier_copy
+        self.data_stream = data_stream
+        self.fit_kwargs = fit_kwargs
+        self.verbose = verbose
+        self.test_data = test_data
+
+        self.iterations = 0
+        self.n_samples = 0
+
+        if isinstance(self.classifier_copy.copy, Pipeline):
+            self.is_pipeline = True
+        else:
+            self.is_pipeline = False
+
+    def run(self):
+        """Function to start the copying of the continuous online classifier."""
+        self.continuous_copy_run = True
+
+        self.partial_fit_ok = check_partial_fit(self.classifier_copy.copy)
+        if not self.partial_fit_ok:
+            print("Sorry, not all those transformers/estimators have 'partial_fit'.")
```

Comment:
I've removed self.partial_fit_ok from the while loop to simplify. However, I'd like to keep it so I can distinguish from the situation where the thread is stopped externally or otherwise and where it can't start due to this reason, so it displays the helper message only on this situation.